api/envoy/api/v2/route/route.proto

```diff
@@ -939,6 +939,22 @@ message HeaderMatcher {
     // If specified, header match will be be performed based on whether the header is in the
     // request.
     bool present_match = 7;
+
+    // If specified, header match will be performed based on the prefix of the header value.
+    //
+    // Examples:
+    //
+    // * The prefix *abcd* matches the value *abcdxyz*, but not for *abcxyz*.
+    // * The empty prefix is also valid and matches any value.
```

Comment:
we already have `present_match` above, I don't think empty prefix adds value. (same for suffix_match)

Reply:
Yeah, I just want to make it clear in the comment that an empty prefix/suffix will match to anything which is how people usually would expect for prefix/suffix match.
Do you want to explicitly disallow empty string here? I feel it's no harm to provide more flexibility to user.

Reply:
I would probably disallow it using a constraint. IMO there is no reason to allow multiple ways of doing the same thing within the same config structure.